GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRMouseDeviceManager.java

```diff
@@ -75,9 +80,17 @@ void removeCursorController(GVRBaseController controller) {
             thread.quitSafely();
             thread = new EventHandlerThread(THREAD_NAME);
             threadStarted = false;
+            if(gvrContext != null){
+                gvrContext.unregisterDrawFrameListener(this);
+            }
         }
     }
 
+    @Override
+    public void onDrawFrame(float frameTime) {
+        thread.invalidate();
```

Comment:
@liaxim the drawframelistener in my case does exactly (2).

It does very little and only notifies the thread to invalidate the position.